src/scipp/scipy/optimize/__init__.py

```diff
@@ -24,7 +25,20 @@ def _as_scalar(obj, unit):
     return scalar(value=obj, unit=unit)
 
 
-def _wrap_func(f, p_names, p_units):
+def _wrap_func2(f, p_names, p_units, x_units):
+    def func(x, *args):
+        if len(x.shape) == 1:
+            x = x.reshape(1, -1)
+        xs = [
+            array(dims=('row',), unit=unit, values=xi) for unit, xi in zip(x_units, x)
```

Comment:
This looks inefficient, why do we need to make a new variable (and thus copy/allocate memory) every time?

Reply:
Hm, we could probably create the scipp arrays once outside of the fit function and only replace the `.values` here instead.

Reply:
Not recreating the arrays did make it about 20% faster.
But it seems most of the overhead is elsewhere.

Did some profiling:

## For 1D fit for all combinations of 2 coordinates (e.g. many small fits)
**Total runtime (scipp):** 1.58 s
**Total runtime (xarray):** 189 ms
**Time spent in fit function (f) (scipp):** 750ms (50%)
**Time spent in fit function (f) (xarray):** 47ms (25%)
**Time spent in `sc.compat.to_xarray`: <1ms
**Time spent in `curve_fit` outside of the call to `xarray.Dataset.curvefit`: <1%.

Some conslusions:
    * That the time spent in the fit function is larger for scipp is not surprising. It has more overhead than the pure Numpy fit function, and the arrays we are dealing with are small. I assume this can not be improved much?
    * The difference between xarray and scipp outside of the fit functions is more difficult to explain.
    * As long as the fit function consumes 50% of the runtime any improvements outside of it will have marginal impact on total runtime.